include/log.h

```diff
@@ -67,6 +67,8 @@ struct ofono_debug_desc {
 					__FILE__, __FUNCTION__ , ## arg); \
 } while (0)
 
+#define PRINTABLE_STR(s) ((s) ? (s) : "(null)")
```

Comment:
Try

volatile char *str = NULL;
printf("%s", str);

for instance. The key here is that glibc actually calls puts() directly in this case. With printf("str: %s", str), it does not crash.